accord-core/src/main/java/accord/messages/ReadData.java

```diff
@@ -251,9 +314,9 @@ private void read(SafeCommandStore safeStore, Command.Committed command)
 
     void obsolete()
     {
-        if (!isObsolete)
+        if (state == State.PENDING)
```

Comment:
Does this also have an illegal state we could check for if we try to mark it as obsolete after already having sent the ack?

Reply:
if its `RETURNED` then that's fine, it just means that the txn was invalidated or applied after the fact.

If its OBSOLETE that just means another store set it

so both cases should be fine to ignore